File: arena-jface/src/main/java/org/uqbar/lacar/ui/impl/jface/lists/JFaceListBuilder.java
package org.uqbar.lacar.ui.impl.jface.lists;

import org.eclipse.jface.viewers.ListViewer;
import org.eclipse.swt.SWT;
import org.eclipse.swt.widgets.Composite;
import org.eclipse.swt.widgets.List;
import org.uqbar.lacar.ui.impl.jface.JFaceContainer;

public class JFaceListBuilder<T> extends JFaceAbstractListBuilder<T, ListViewer, List> {
	
	public JFaceListBuilder(JFaceContainer container) {
		super(container);
	}

	protected ListViewer createViewer(Composite jFaceComposite) {
		ListViewer viewer = new ListViewer(jFaceComposite, SWT.SINGLE | SWT.BORDER | SWT.FULL_SELECTION);
		this.initialize(viewer.getList());
		return viewer;
	}
}

File: arena-jface/src/main/scala/org/uqbar/lacar/ui/impl/jface/builder/lists/JFaceListBuilder.scala
package org.uqbar.lacar.ui.impl.jface.builder.lists

import org.eclipse.jface.viewers.ListViewer
import org.eclipse.swt.widgets.List
import org.uqbar.lacar.ui.impl.jface.JFaceContainer
import org.eclipse.swt.widgets.Composite
import org.eclipse.swt.SWT._

class JFaceListBuilder[T](container:JFaceContainer) extends JFaceAbstractListBuilder[T, ListViewer, List] (container) {

  override def createViewer(jFaceComposite:Composite) = {
		val viewer = new ListViewer(jFaceComposite, H_SCROLL | V_SCROLL | SINGLE | BORDER | FULL_SELECTION)
		initialize(viewer.getList)
		viewer
	}
  
}